app/src/main/java/io/apicurio/registry/storage/impl/sql/AbstractSqlRegistryStorage.java

```diff
@@ -2660,17 +2703,19 @@ protected void importArtifactVersion(Handle handle, ArtifactVersionEntity entity
             }
         } else {
             log.info("Duplicate globalId detected, skipping import of artifact version.");
+            //TODO generate a new globalId and add the artifact, maybe only depending on a feature flag
```

Comment:
+1 to the feature flag to decide what to do here, but that's work for a different PR.